### PR TITLE
move to preact

### LINKED ISSUE #1: Refactor to use React
I've originally done this as a personal challenge to not use it, and it was wonderful to learn how much easier React makes our lives.

### LINKED ISSUE #4: 🐛 Fix known bugs
This is an umbrella issue for some problems we've encountered in the current tool.

- [x] Looks like the logo svg sometimes bug when exporting the image
screen shot 2018-10-16 at 10 58 52

- [ ] There are indentation issues on the "css" textbox.
> I guess #1 might fix this as a side effect.

### PR DESCRIPTION
**Pull request by leonardodino**
# very refactor, much wow  

screen shot 2018-10-18 at 11 09 28 am

closes #1 closes #4 

probably opens a bunch of others.
happy `🎃 hacktoberfest`! 

## features
- [x] fixed output size (facebook)
- [x] preact, 8kb gzipped bundle total
- [x] "fancier" style, added footer
- [x] works offline (or should, idk)

## regressions
- [ ] removed syntax-highlighting
- [ ] _maybe_ it does not work on `gh-pages`
  - i haven't found a way to configure the `output.publicPath`
  - maybe migrate to netlify? bonus: stop littering the repo with build artifacts

## pictures
|  |  |  |
|:---:|:---:|:---:|
| [`web design`](https://plant-zipper.surge.sh) | `bundle size` | [`visual diff`](https://www.facebook.com/codesigncommunity/photos/gm.2122473581327899/2647979432094016/?type=3&theater) |

### CODE REVIEW

**Comment by leonardodino** [comment]
tried to follow the current structure as close as possible:

**components**
- `App`
- `ControlPanel`
- `EventCover`

**state**
- `color`
- `number`
- `body`

changed how the image is generated, but the code part is still rendered with HTML.

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,5 @@
-.DS_Store
-dist
-.cache
 node_modules
+/build
+/*.log
+.*cache
+package-lock.json
```
Comment (Yurickh) [inline .gitignore]: The lock file should always me committed alongside the rest of the code.

Comment (Yurickh) [reply .gitignore]: Since we're using yarn, the mere presence of a package-lock is a smell and git can make sure it is put into evidence.

Comment (leonardodino) [reply .gitignore]: the idea is reducing friction on contribution for people not using yarn.
as long as they don't install any new package, it should be safe.

that, coupled with netlify, as a CI/CD, solves much of the dependency version mismatch issue.

Comment (leonardodino) [reply .gitignore]: as it doesn't interfere with my tools, i have no trouble in removing it the `.gitignore`. 

(:

Comment (Yurickh) [reply .gitignore]: I would advocate for removing it, sure.
We can debate the merits of it in the next co-de-sign haha.

Comment (leonardodino) [reply .gitignore]:
```suggestion
```

Comment (leonardodino) [reply .gitignore]: screen shot 2018-10-18 at 11 47 07 am
github suggestions 👌

**Comment by Yurickh** [comment]
Can you ping me once you're done with this PR?

**Comment by leonardodino** [comment]
sorry, I got busy, stopped working on it :/

feel free to improve it, or take another approach rewriting (:
I've done it mostly as a personal experimentation with `preact`.

the only critical part missing is `syntax-highlighting`.

(being only css i know it's not-hard to write a parser for, even from the ground-up. it might even be a good exercise (: but I will only have any free-time after mid-november)

**Comment by Yurickh** [comment]
I'll close this PR then, we can re-open it once you get back working, if there's no one else working on different alternatives :)~

**Pull request closed without merging**